src/network/networkd-address.c

```diff
@@ -931,8 +931,6 @@ static int address_handler(sd_netlink *rtnl, sd_netlink_message *m, Link *link)
         assert(link);
         assert(link->ifname);
         assert(link->address_messages > 0);
-        assert(IN_SET(link->state, LINK_STATE_CONFIGURING,
-               LINK_STATE_FAILED, LINK_STATE_LINGER));
 
         link->address_messages--;
 
```

Comment:
Shouldn't the condition below be inverted? I.e. only do further processing only when `link->state` is `LINK_STATE_CONFIGURING`? Right now potentially we'd do that also on `LINK_STATE_UNMANAGED` and others.

Reply:
Yeah. But, I'd like to leave that to another PR. Currently, `link->state` conditions are borked in many places. I'd like to rewrite related code comprehensively later.